### PR TITLE
wizer: Partially support nested components

### PR DESCRIPTION
**Pull request by alexcrichton**
Support nested components insofar as they don't have access to modules at this time. This helps `wasmtime wizer` support wasm-tools-generated components which use nested components for type-ascribing exports. Otherwise though nested components are not allowed to defined or instantiate core modules as that would need more implementation in Wizer to support.

### CODE REVIEW

File: crates/wizer/tests/all/component.rs [hunk 1]
```diff
@@ -301,6 +336,29 @@ async fn snapshot_memory() -> Result<()> {
...
+                (func (export "run") (result i32) i32.const 42)
+            )
+            (core instance $i (instantiate $m))
+            (func (export "run") (result u32) (canon lift (core func $i "run")))
+            (func (export "wizer-initialize") (canon lift (core func $i "init")))
+        )"#,
+    )
+    .await?;
+
+    Ok(())
```
Comment (fitzgen) [inline crates/wizer/tests/all/component.rs:381]: Maybe also add a test with a nesting level greater than the minimum two?
